snappi_ixnetwork/trafficitem.py

```diff
@@ -191,9 +220,245 @@ class TrafficItem(CustomField):
+                url = "%s/traffic/trafficItem" % self._api._ixnetwork.href
+                self._api._request("DELETE", url)
+                self._api._ixnetwork.Traffic.TrafficItem.find().refresh()
+        self.traffic_index = 1
+
+    def create_traffic(self, config):
+        flows = config.flows
+        tr = {"xpath": "/traffic", "trafficItem": []}
+        ports = self.get_ports_xpath(config)
+        devices = self.get_device_xpath(config)
+        for i, f in enumerate(flows):
+            if f._properties.get("name") is None:
+                raise Exception("name shall not be null for flows")
+            if f._properties.get("tx_rx") is None:
+                msg = (
+                    "Please configure the flow endpoint"
+                    "for flow indexed at %s" % i
+                )
+                raise Exception(msg)
+            if f.tx_rx.choice is None:
+                msg = "Flow endpoint needs to be either port or device"
+                raise Exception(msg)
+            if f.tx_rx.choice == "port":
+                tr_type = "raw"
+                ep = getattr(f.tx_rx, "port")
+                tx_objs = ["%s/protocols" % ports.get(ep.tx_name)]
+                rx_objs = ["%s/protocols" % ports.get(ep.rx_name)]
+            else:
+                ep = getattr(f.tx_rx, "device")
+                tr_type = devices[ep.tx_names[0]]["type"]
+                tx_objs = [devices[n]["xpath"] for n in ep.tx_names]
+                rx_objs = [devices[n]["xpath"] for n in ep.rx_names]
+            tr_xpath = "/traffic/trafficItem[%d]" % self.traffic_index
+            tr["trafficItem"].append(
+                {"xpath": tr_xpath, "name": "%s" % f.name}
+            )
+
+            tr["trafficItem"][-1]["endpointSet"] = [
+                {"xpath": tr["trafficItem"][-1]["xpath"] + "/endpointSet[1]"}
+            ]
+            tr["trafficItem"][-1]["endpointSet"][0]["sources"] = [
+                o for o in tx_objs
+            ]
+            tr["trafficItem"][-1]["endpointSet"][0]["destinations"] = [
+                o for o in rx_objs
+            ]
+            tr["trafficItem"][-1]["trafficType"] = tr_type
+            if tr_type == "raw":
+                tr["trafficItem"][-1]["configElement"] = self.config_raw_stack(
+                    tr_xpath, f.packet
+                )
+            self.traffic_index += 1
+        return tr
+
+    def config_raw_stack(self, xpath, packet):
+        ce_path = "%s/configElement[1]" % xpath
+        config_elem = {"xpath": ce_path, "stack": []}
+        for i, header in enumerate(packet):
+            stack_name = self._HEADER_TO_TYPE.get(header._choice)
+            header_xpath = "%s/stack[@alias = '%s-%d']" % (
+                ce_path,
+                stack_name,
+                i + 1,
+            )
+            self._append_header(header, header_xpath, config_elem["stack"])
+        fcs_xpath = "%s/stack[@alias = '%s-%d']" % (
```

Comment:
This is not valid if stack do not have any Ethernet stack.(say PfcPause)

Reply:
Agree. need to handle for pfcpause only. for ethernet or ethenetpause fcs will be same.
I think Ixnetwork has only these 2 l2 headers. correct me if I am wrong